ff/serving/job/materialize_job_test.go

```diff
@@ -0,0 +1,239 @@
+		}
+	}
+	return nil
+}
+
+type CopyTestData struct {
+	Rows []interface{}
+}
+
+func TestJobs(t *testing.T) {
+	emptyList := CopyTestData{
+		Rows: []interface{}{},
+	}
+	basicNumList := CopyTestData{
+		Rows: []interface{}{1, 2, 3, 4, 5},
+	}
+
+	stringNumList := CopyTestData{
+		Rows: []interface{}{"one", "two", "three", "four", "five"},
+	}
+	multipleTypesList := CopyTestData{
+		Rows: []interface{}{1, "two", 3.0, 'f', false},
+	}
+
+	numListofLists := CopyTestData{
+		Rows: []interface{}{[]int{1, 2, 3}, []int{2, 3, 4}, []int{3, 4, 5}},
+	}
+
+	differentTypeLists := CopyTestData{
+		Rows: []interface{}{[]int{1, 2, 3}, []string{"two", "three", "four"}, []float64{3.0, 4.0, 5.0}},
+	}
+	testJobs := []JobTestParams{
+		JobTestParams{
+			TestName:         "Basic copy test",
+			FeatureTableData: basicNumList.Rows,
+			ChunkSize:        5,
+			ChunkIdx:         0,
+		},
+		JobTestParams{
+			TestName:         "Partial copy test",
+			FeatureTableData: basicNumList.Rows,
+			ChunkSize:        2,
+			ChunkIdx:         0,
+		},
+		JobTestParams{
+			TestName:         "Chunk size overflow test",
+			FeatureTableData: basicNumList.Rows,
+			ChunkSize:        6,
+			ChunkIdx:         0,
+		},
+		JobTestParams{
+			TestName:         "Single copy test",
+			FeatureTableData: basicNumList.Rows,
+			ChunkSize:        1,
+			ChunkIdx:         0,
+		},
+		JobTestParams{
+			TestName:         "Final index copy test",
+			FeatureTableData: basicNumList.Rows,
+			ChunkSize:        1,
+			ChunkIdx:         4,
+		},
+		JobTestParams{
+			TestName:         "Last overlap chunk test",
+			FeatureTableData: basicNumList.Rows,
+			ChunkSize:        2,
+			ChunkIdx:         2,
+		},
+		JobTestParams{
+			TestName:         "Zero chunk size copy test",
+			FeatureTableData: basicNumList.Rows,
+			ChunkSize:        0,
+			ChunkIdx:         0,
+		},
+		JobTestParams{
+			TestName:         "String list copy test",
+			FeatureTableData: stringNumList.Rows,
+			ChunkSize:        5,
+			ChunkIdx:         0,
+		},
+		JobTestParams{
+			TestName:         "Different types copy test",
+			FeatureTableData: multipleTypesList.Rows,
+			ChunkSize:        5,
+			ChunkIdx:         0,
+		},
+		JobTestParams{
+			TestName:         "List features test",
+			FeatureTableData: numListofLists.Rows,
+			ChunkSize:        5,
+			ChunkIdx:         0,
+		},
+		JobTestParams{
+			TestName:         "List features different types",
+			FeatureTableData: differentTypeLists.Rows,
+			ChunkSize:        5,
+			ChunkIdx:         0,
+		},
+		JobTestParams{
+			TestName:         "No rows test",
+			FeatureTableData: emptyList.Rows,
+			ChunkSize:        1,
+			ChunkIdx:         0,
+		},
+		JobTestParams{
+			TestName:         "No rows/zero chunk size test",
+			FeatureTableData: emptyList.Rows,
+			ChunkSize:        0,
+			ChunkIdx:         0,
+		},
+	}
+	for _, param := range testJobs {
+		err := testParams(param)
```

Comment:
fit this into the if statement below

Reply:
What do I fit into the if statement?

Reply:
if err := testParams(param); err != nil {

}